modules/candidate_list/jsx/CandidateListFilter.js

```diff
@@ -0,0 +1,293 @@
+import Panel from 'Panel';
+import FilterForm from 'FilterForm';
+
+class CandidateListFilter extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.getBasicOpts = this.getBasicOpts.bind(this);
+    this.getAdvancedOpts = this.getAdvancedOpts.bind(this);
+    this.updateFilter = this.updateFilter.bind(this);
+    this.resetFilters = this.resetFilters.bind(this);
+    this.toggleAdvanced = this.toggleAdvanced.bind(this);
+
+    this.getDisplayedFormElements = this.getDisplayedFormElements.bind(this);
+    this.resetFilters = this.resetFilters.bind(this);
+    this.componentDidMount = this.componentDidMount.bind(this);
+    this.state = {
+        // Whether or not to show the advanced options
+        showadvanced: false,
+        dynamicfilters : {
+            projects : { '' : 'All', 'placeholder' : 'Loading valid projects..' },
+            subprojects : { '' : 'All', 'placeholder' : 'Loading valid subprojects..' }, 
+            visits : { '' : 'All', 'placeholder' : 'Loading valid visits...' }, 
+            sites : { '' : 'All', 'placeholder' : 'Loading valid sites...' },
+            participantstatus : { '' : 'All', 'placeholder' : 'Loading participant status options...' },
+        },
+        filter: {}
+    };
+  }
+  updateFilter(filter) {
+	// FIXME: This should be this.props instead of this.state, but for now
+	// the caller and callee are formatted differently, so just use a callback
+    if (filter.ParticipantStatus) {
+        filter.ParticipantStatus.exactMatch = true;
+    }
+    this.setState({filter});
+	if(this.props.onFilterUpdated) {
+		this.props.onFilterUpdated(filter);
+	}
+  }
+
+  getBasicOpts() {
+      var basicopts = {
```

Comment:
@driusan Is there something wrong with LorisForm.php?

Reply:
I don't see why non-dynamic display things would be in PHP and not the react component, and adding display things to the toJSON doesn't make sense to me.

This is mostly an attempt to try a different, more independent model. I'd also like to try one that uses {this.props.children} somewhere so that the three ways can be compared.

It also needs to handle the Advanced/Basic views dynamically, which can't be done by LorisForm.php